cmd/titus-log-uploader/config.go

```diff
@@ -0,0 +1,164 @@
+package main
+
+import (
+	"os"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/Netflix/titus-executor/filesystems"
+)
+
+const (
+	envKeyLocalDir            = "WATCH_LOCAL_DIR"
+	envKeyUploadDir           = "WATCH_UPLOAD_DIR"
+	envKeyUploadRegexp        = "WATCH_UPLOAD_REGEXP"
+	envKeyUploadCheckInterval = "WATCH_UPLOAD_CHECK_INTERVAL"
+	envKeyUploadThresholdTime = "WATCH_UPLOAD_THRESHOLD_TIME"
+	envKeyStdioCheckInterval  = "WATCH_STDIO_CHECK_INTERVAL"
+	envKeyKeepFileAfterUpload = "WATCH_KEEP_FILE_AFTER_UPLOAD"
+
+	envKeyBucketName     = "UPLOAD_BUCKET_NAME"
+	envKeyPathPrefix     = "UPLOAD_PATH_PREFIX"
+	envKeyTaskRole       = "UPLOAD_TASK_ROLE"
+	envKeyTaskID         = "UPLOAD_TASK_ID"
+	envKeyWriterRole     = "UPLOAD_WRITER_ROLE"
+	envKeyUseDefaultRole = "UPLOAD_USE_DEFAULT_ROLE"
+)
+
+func mustReadStringFromEnv(envKey string) string {
+	if val, ok := os.LookupEnv(envKey); ok {
+		return val
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"variable": envKey,
+		"type":     "string",
+	}).Fatal("Missing required environment variable")
+
+	return ""
+}
+
+func mustReadDurationFromEnv(envKey string) time.Duration {
+	if val, ok := os.LookupEnv(envKey); ok {
+		dur, err := time.ParseDuration(val)
+		if err != nil {
+			logrus.WithFields(logrus.Fields{
+				"key":   envKey,
+				"value": val,
+				"err":   err,
+			}).Fatal("Unable to parse valid duration from environment variable")
+			return 0
+		}
+		return dur
+	}
+
+	logrus.WithFields(logrus.Fields{
+		"variable": envKey,
+		"type":     "duration",
+	}).Fatal("Missing required environment variable")
+	return 0
+}
+
+func mustReadRegexpFromEnv(envKey string) *regexp.Regexp {
+	if val, ok := os.LookupEnv(envKey); ok {
+		r, err := regexp.Compile(val)
+		if err != nil {
+			logrus.WithFields(logrus.Fields{
+				"key":   envKey,
+				"value": val,
+				"err":   err,
+			}).Error("Unable to parse valid regular expression from environment variable")
```

Comment:
I haven't seen this `mustFoo` pattern before.

Is there a particular reason for handling errors like this and continuing in the face of these errors?

Reply:
Will fix this error to be a `Fatal()` and on line 94 as well.  Is your preference to Panic?  I was modeling off of the `template.Must` from the stdlib: https://golang.org/pkg/html/template/#Must

I've seen it elsewhere but examples are, of course, eluding me.